packages/grub/0047-Revert-UBUNTU-Move-verifiers-after-decompressors.patch

```diff
@@ -0,0 +1,48 @@
+From 95cafe6cf7dd2a02bd33ddee624bb9b7c3a931ae Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Tue, 13 Feb 2024 22:21:41 +0000
+Subject: [PATCH] Revert "UBUNTU: Move verifiers after decompressors"
+
+We use the PGP verifier to validate the signature of grub.cfg, and do
+not want to expose the decompressors to untrusted input.
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ include/grub/file.h        | 4 ++--
+ tests/file_filter/test.cfg | 2 +-
+ 2 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/include/grub/file.h b/include/grub/file.h
+index fa23688..96827a4 100644
+--- a/include/grub/file.h
++++ b/include/grub/file.h
+@@ -180,13 +180,13 @@ extern grub_disk_read_hook_t EXPORT_VAR(grub_file_progress_hook);
+ /* Filters with lower ID are executed first.  */
+ typedef enum grub_file_filter_id
+   {
++    GRUB_FILE_FILTER_VERIFY,
+     GRUB_FILE_FILTER_GZIO,
+     GRUB_FILE_FILTER_XZIO,
+     GRUB_FILE_FILTER_LZOPIO,
++    GRUB_FILE_FILTER_MAX,
+     GRUB_FILE_FILTER_COMPRESSION_FIRST = GRUB_FILE_FILTER_GZIO,
+     GRUB_FILE_FILTER_COMPRESSION_LAST = GRUB_FILE_FILTER_LZOPIO,
+-    GRUB_FILE_FILTER_VERIFY,
+-    GRUB_FILE_FILTER_MAX,
```

Comment:
unnecessary sort?

Reply:
No, the order does matter here. Enumerations in C are numeric, not symbolic, i.e. the order changes their assigned values. It is idiomatic to loop e.g. from 0 to some sentinel value (like `GRUB_FILE_FILTER_MAX` here). For this particular case, the ordering of filters in the code determines their numeric value and therefore in which order they're applied. In any case, those two patches are reverts of downstream patches packaged in the SRPM and hence should match what's in there.